src/components/contribute/carousel/information-modal.tsx

```diff
@@ -38,13 +42,26 @@ interface InformationModalProps {
     onRequestClose: () => void;
 }
 
-export class InformationModal extends React.Component<InformationModalProps> {
-    constructor(props: InformationModalProps) {
+type Props = InformationModalProps & ReturnType<typeof mapStateToProps>;
+
+class Information extends React.Component<Props> {
+    constructor(props: Props) {
         super(props);
     }
 
+    getModalMessage = (contributeType: string | undefined) => {
+        switch (contributeType) {
+            case ContributeType.SPEAK:
+                return <TipsSpeak />;
+            case ContributeType.LISTEN:
+                return <TipsVerify />;
+            case ContributeType.REPEAT:
+                return <TipsSpeak />;
```

Comment:
Is there no TipsRepeat or something? I assume that, for repeat, the same modal message is supposed to be displayed as in speak?

Reply:
Correct! Herma is now using the Speak tips, they are similar enough. But maybe it would be worth adding some tips for what to do when the repeated clip is bad.